drivers/SmartThings/zwave-switch/src/zooz-zen-30-dimmer-relay/init.lua

```diff
@@ -0,0 +1,151 @@
+  end
+  return false
+end
+
+local BUTTON_VALUES = {
+  "up_hold", "down_hold", "held",
+  "up","up_2x","up_3x","up_4x","up_5x",
+  "down","down_2x","down_3x","down_4x","down_5x",
+  "pushed", "pushed_2x","pushed_3x","pushed_4x","pushed_5x"
+}
+
+local map_key_attribute_to_capability = {
+  [CentralScene.key_attributes.KEY_PRESSED_1_TIME] = {
+    [0x01] = capabilities.button.button.up(),
+    [0x02] = capabilities.button.button.down(),
+    [0x03] = capabilities.button.button.pushed()
+  },
+  [CentralScene.key_attributes.KEY_PRESSED_2_TIMES] = {
+    [0x01] = capabilities.button.button.up_2x(),
+    [0x02] = capabilities.button.button.down_2x(),
+    [0x03] = capabilities.button.button.pushed_2x()
+  },
+  [CentralScene.key_attributes.KEY_PRESSED_3_TIMES] = {
+    [0x01] = capabilities.button.button.up_3x(),
+    [0x02] = capabilities.button.button.down_3x(),
+    [0x03] = capabilities.button.button.pushed_3x()
+  },
+  [CentralScene.key_attributes.KEY_PRESSED_4_TIMES] = {
+    [0x01] = capabilities.button.button.up_4x(),
+    [0x02] = capabilities.button.button.down_4x(),
+    [0x03] = capabilities.button.button.pushed_4x()
+  },
+  [CentralScene.key_attributes.KEY_PRESSED_5_TIMES] = {
+    [0x01] = capabilities.button.button.up_5x(),
+    [0x02] = capabilities.button.button.down_5x(),
+    [0x03] = capabilities.button.button.pushed_5x()
+  },
+  [CentralScene.key_attributes.KEY_HELD_DOWN] = {
+    [0x01] = capabilities.button.button.up_hold(),
+    [0x02] = capabilities.button.button.down_hold(),
+    [0x03] = capabilities.button.button.held()
+  }
+}
+
+local function endpoint_to_component(device, endpoint)
+  if endpoint == 1 then
+    return "switch1"
+  elseif endpoint == 0 then
+    return "main"
+  end
+end
+
+local function component_to_endpoint(device, component)
+  if component == "switch1" then
+    return {1}
+  elseif component == "main" then
+    return {0}
+  end
+end
+
+local function central_scene_notification_handler(driver, device, cmd)
+  if(cmd.args.key_attributes == 0x01) then
+    log.error("Button Value 'released' is not supported by SmartThings")
+    return
+  end
+
+  if device:get_field(LAST_SEQ_NUMBER_KEY) ~= cmd.args.sequence_number then
+    device:set_field(LAST_SEQ_NUMBER_KEY, cmd.args.sequence_number)
+    local event_map = map_key_attribute_to_capability[cmd.args.key_attributes]
+    local event = event_map and event_map[cmd.args.scene_number]
+    if event ~= nil then
+      device:emit_event_for_endpoint("main", event)
+    end
+  end
+end
+
+local function added_handler(self, device)
+  device:emit_event(capabilities.button.supportedButtonValues(BUTTON_VALUES))
+  device:emit_event(capabilities.button.numberOfButtons({value = 3}))
+end
+
+local function device_init(self, device)
+  device:set_field(LAST_SEQ_NUMBER_KEY, -1)
+  device:set_endpoint_to_component_fn(endpoint_to_component)
+  device:set_component_to_endpoint_fn(component_to_endpoint)
+end
+
+local function switch_level_set(driver, device, cmd)
```

Comment:
did you redefine this because of the behavior around 0?

Reply:
> did you redefine this because of the behavior around 0?

yes

Reply:
We're having a discussion on this for all dimmer switches, so please remove this change for now.

Reply:
done